From c6e2ecacb40767e3327813808c60653181e32f78 Mon Sep 17 00:00:00 2001
From: Ronan Pigott <ronan@rjp.ie>
Date: Wed, 11 Dec 2024 12:47:10 -0700
Subject: [PATCH] manager: s/deserialized_subscribed/subscribed_as_strv

Now that this field may get populated at runtime, the deserialized name
is misleading. Change the name to reflect its updated purpose.

(cherry picked from commit e1315a621ae26473fcc9cd0d6013836f5f498d40)

Resolves: RHEL-73780
---
 src/core/dbus.c              | 6 +++---
 src/core/manager-serialize.c | 2 +-
 src/core/manager.c           | 6 +++---
 src/core/manager.h           | 2 +-
 4 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/src/core/dbus.c b/src/core/dbus.c
index 9abc35888d..58cd1ee175 100644
--- a/src/core/dbus.c
+++ b/src/core/dbus.c
@@ -860,8 +860,8 @@ int bus_init_api(Manager *m) {
         if (r < 0)
                 return log_error_errno(r, "Failed to set up API bus: %m");
 
-        (void) bus_track_coldplug(bus, &m->subscribed, /* recursive= */ false, m->deserialized_subscribed);
-        m->deserialized_subscribed = strv_free(m->deserialized_subscribed);
+        (void) bus_track_coldplug(bus, &m->subscribed, /* recursive= */ false, m->subscribed_as_strv);
+        m->subscribed_as_strv = strv_free(m->subscribed_as_strv);
         m->api_bus = TAKE_PTR(bus);
 
         return 0;
@@ -1013,7 +1013,7 @@ static void destroy_bus(Manager *m, sd_bus **bus) {
                 r = bus_track_to_strv(m->subscribed, &subscribed);
                 if (r < 0)
                         log_warning_errno(r, "Failed to serialize api subscribers, ignoring: %m");
-                strv_free_and_replace(m->deserialized_subscribed, subscribed);
+                strv_free_and_replace(m->subscribed_as_strv, subscribed);
 
                 m->subscribed = sd_bus_track_unref(m->subscribed);
         }
diff --git a/src/core/manager-serialize.c b/src/core/manager-serialize.c
index 3f624619df..bc29ac91c6 100644
--- a/src/core/manager-serialize.c
+++ b/src/core/manager-serialize.c
@@ -493,7 +493,7 @@ int manager_deserialize(Manager *m, FILE *f, FDSet *fds) {
                         (void) exec_shared_runtime_deserialize_one(m, val, fds);
                 else if ((val = startswith(l, "subscribed="))) {
 
-                        r = strv_extend(&m->deserialized_subscribed, val);
+                        r = strv_extend(&m->subscribed_as_strv, val);
                         if (r < 0)
                                 return r;
                 } else if ((val = startswith(l, "varlink-server-socket-address="))) {
diff --git a/src/core/manager.c b/src/core/manager.c
index 068ea2ab03..4a7132274f 100644
--- a/src/core/manager.c
+++ b/src/core/manager.c
@@ -1804,7 +1804,7 @@ Manager* manager_free(Manager *m) {
         free(m->switch_root_init);
 
         sd_bus_track_unref(m->subscribed);
-        strv_free(m->deserialized_subscribed);
+        strv_free(m->subscribed_as_strv);
 
         unit_defaults_done(&m->defaults);
 
@@ -2142,10 +2142,10 @@ int manager_startup(Manager *m, FILE *serialization, FDSet *fds, const char *roo
                 manager_setup_bus(m);
 
                 /* Now that we are connected to all possible buses, let's deserialize who is tracking us. */
-                r = bus_track_coldplug(m->api_bus, &m->subscribed, false, m->deserialized_subscribed);
+                r = bus_track_coldplug(m->api_bus, &m->subscribed, false, m->subscribed_as_strv);
                 if (r < 0)
                         log_warning_errno(r, "Failed to deserialized tracked clients, ignoring: %m");
-                m->deserialized_subscribed = strv_free(m->deserialized_subscribed);
+                m->subscribed_as_strv = strv_free(m->subscribed_as_strv);
 
                 r = manager_varlink_init(m);
                 if (r < 0)
diff --git a/src/core/manager.h b/src/core/manager.h
index e4cada80ff..7016eab2d3 100644
--- a/src/core/manager.h
+++ b/src/core/manager.h
@@ -340,7 +340,7 @@ struct Manager {
         considered subscribes, since they last for very short only,
         and it is much simpler that way. */
         sd_bus_track *subscribed;
-        char **deserialized_subscribed;
+        char **subscribed_as_strv;
 
         /* This is used during reloading: before the reload we queue
          * the reply message here, and afterwards we send it */
